data_steward/utils/pipeline_logging.py

```diff
@@ -111,3 +159,86 @@ def setup_logger(log_filepath_list, console_logging=True):
     for filename in log_path:
         log_list.append(create_logger(filename, console_logging))
     return log_list
+
+
+def _get_date_str():
+    """
+    Get current date formatted using FILENAME_FMT
+    :return: 
+    """
+    return datetime.today().strftime(FILENAME_FMT)
+
+
+def _get_log_file_path():
+    """
+    Get the abs path of the log file to use. 
+    
+    The location is DEFAULT_LOG_DIR and the file name is the current
+    date formatted using FILENAME_FMT.
+
+    :return: absolute path to the log file
+    """
+    date_str = _get_date_str()
+    return os.path.join(DEFAULT_LOG_DIR, f'{date_str}.log')
+
+
+def _get_config(filename, level):
+    """
+    Get a dictionary which describes the logging configuration
+    
+    :param filename: Create the FileHandler using the specified filename.
+    :param level: Set the root logger level to the specified level 
+                  (i.e. logging.{DEBUG,INFO,WARNING,ERROR}).
+    :return: the default configuration dict
+    """
+    # copy _DEFAULT_CONFIG to avoid modifying it
+    default_config = dict(_DEFAULT_CONFIG)
+    default_config['root']['level'] = level
+    file_log_handler_dict = default_config['handlers'][_FILE_HANDLER]
+    file_log_handler_dict['filename'] = filename
+    return default_config
+
+
+def _except_hook(exc_type, exc_value, exc_traceback):
+    """
+    Log exception info to root logger. Used as a hook for uncaught exceptions prior 
+    to system exit.
+    
+    :param exc_type: type of the exception
+    :param exc_value: the exception
+    :param exc_traceback: the traceback associated with the exception
+    """
+    root_logger = logging.getLogger()
+    root_logger.critical('Uncaught exception',
+                         exc_info=(exc_type, exc_value, exc_traceback))
+
+
+def configure(level=logging.INFO):
+    """
+    Configure the logging system for use by pipeline.
+    
+    Creates a handler which writes to sys.stderr and a handler which appends to a file 
+    named according to the current date. Both handlers' formattters are set 
+    using the LOG_FORMAT format string and are added to the root logger.
+    
+    :param level: Set the root logger level to the specified level (i.e. 
+                  logging.{DEBUG,INFO,WARNING,ERROR}), defaults to INFO.
+    
+    :example:
+    >>> from utils import pipeline_logging
+    >>> 
+    >>> LOGGER = logging.getLogger(__name__)
+    >>>
+    >>> def func(p1):
+    >>>     LOGGER.debug(f"func called with p1={p1}")
+    >>>     LOGGER.info("func called")
+    >>>
+    >>> if __name__ == '__main__':
+    >>>     pipeline_logging.configure()
+    >>>     func(1, 2)
+    """
+    filename = _get_log_file_path()
+    default_config = _get_config(filename, level)
```

Comment:
These lines could be altered.  The `_get_config(level, console_logging)` function should contain the call to `_get_log_file_path()` because we don't let users alter this value.  There's no reason to call it here and pass it as a parameter to another internal function.  We could just call it where needed in the internal function.

```suggestion
    default_config = _get_config(level, console_log)
```

Reply:
Thanks, I've incorporated your suggestion.